pkg/expr/service.go

```diff
@@ -109,6 +109,7 @@ func (s *Service) ExecutePipeline(ctx context.Context, now time.Time, pipeline D
 	for refID, val := range vars {
 		res.Responses[refID] = backend.DataResponse{
 			Frames: val.Values.AsDataFrames(refID),
+			Error: val.Error,
```

Comment:
This is where the main impact on alerting is:

```
vars, err := pipeline.execute(ctx, now, s)
	if err != nil {
		return nil, err
	}
	for refID, val := range vars {
		res.Responses[refID] = backend.DataResponse{
			Frames: val.Values.AsDataFrames(refID),
			Error: val.Error,
```

This is because where before we would return an error from the service (with pipeline execution), we generally are not doing that anymore. Instead, the error moves to the property of each Response, and then gets passed to the caller (alerting).

So alerting needs to check for this condition on evaluation. Also, for the `eval` endpoint, the UI needs to show these errors.

So in short, unless alerting starts checking for the Error on Response(s) - in particular the condition in the case of evaluation, and all expressions for the UI, what was previously shown as errors will become masked.